kratos/containers/csr_matrix.h

```diff
@@ -0,0 +1,463 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//
+
+#if !defined(KRATOS_CSR_MATRIX_H_INCLUDED )
+#define  KRATOS_CSR_MATRIX_H_INCLUDED
+
+
+// System includes
+#include <iostream>
+#include "containers/sparse_contiguous_row_graph.h"
+
+// Project includes
+#include "includes/define.h"
+
+
+namespace Kratos
+{
+///@addtogroup ApplicationNameApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// This function implements "serial" CSR matrix, including capabilities for FEM assembly
+template< class TDataType=double, class TIndexType=std::size_t>
+class CsrMatrix
+{
+public:
+    ///@name Type Definitions
+    ///@{
+    typedef TIndexType IndexType;
+
+    /// Pointer definition of CsrMatrix
+    KRATOS_CLASS_POINTER_DEFINITION(CsrMatrix);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+    CsrMatrix()
+    {
+    }
+
+    /// Default constructor.
+    template<class TGraphType>
+    CsrMatrix(const TGraphType& rSparseGraph)
+    {
+        rSparseGraph.ExportCSRArrays(mRowIndices,mColIndices);
+        mNrows = rSparseGraph.Size();
+
+        //TODO: parallelize
+        IndexType max_col = 0;
+        for(IndexType i=0; i<mColIndices.size(); ++i)
+            max_col = std::max(max_col,mColIndices[i]);
+        mNcols = max_col;
+
+        //initialize mValuesVector to zero
+        mValuesVector.resize(mColIndices.size(),false);
+        SetValue(0.0);
+    }
+
+    /// Destructor.
+    virtual ~CsrMatrix(){}
+
+    /// Assignment operator. TODO: decide if we do want to allow it
+    CsrMatrix& operator=(CsrMatrix const& rOther)=delete;
+    // {
+    //     this->AddEntries(rOther.GetGraph());
+    //     return *this;
+    // }
+
+    ///@}
+    ///@name Operators
+    ///@{
+    void Clear()
+    {
+        mRowIndices.clear();
+        mColIndices.clear();
+        mValuesVector.clear();
+    }
+
+    void SetValue(const TDataType value)
+    {
+        #pragma omp parallel for
+        for(IndexType i=0; i<mValuesVector.size(); ++i)
+            mValuesVector[i] = value;
```

Comment:
this is why the win build fails
I would just use the parallelUtils...

Reply:
agree. done